### PR TITLE
Zorinos support

### LINKED ISSUE #80: OS not supported in Zorin OS
image

I am using Zorin OS (which is based on ubuntu) which is supported but installer is saying otherwise.

Log:
utils/common.sh: line 117: docker: command not found
utils/common.sh: line 120: podman: command not found
/usr/bin/pactl
Operating system not supported.

My system info 
image

### PR DESCRIPTION
**Pull request by suvanbanerjee**
Added zorin os support everything works apart from GUI

Screenshot from 2024-05-10 17-16-54(1)

if choosing GUI getting error with qt5 log https://sprunge.us/b1NHYE
acc to logs qmake is not present but it is 
Screenshot from 2024-05-10 17-19-05(1)

Closes #80

### CODE REVIEW

File: utils/common.sh [hunk 1]
```diff
@@ -361,7 +361,7 @@ function ver() {
 # This function takes an argument like "2f", this will be converted
 # to "0x2f".
 function i2c_get() {
-    if i2cdetect -y -a "$I2C_BUS" "0x$1" "0x$1" 2>>"$LOG_FILE" | grep -q "$1"; then
+    if i2cdetect -y -a "$I2C_BUS" "0x$1" "0x$1" 2>>"$LOG_FILE" | grep -Eq "$1|UU"; then
```
Comment (goldyfruit) [inline utils/common.sh]: This PR should not change anything here, please rebase from master.

File: utils/common.sh [hunk 2]
```diff
@@ -385,9 +385,58 @@ function i2c_scan() {
         for device in "${!SUPPORTED_DEVICES[@]}"; do
             address="${SUPPORTED_DEVICES[$device]}"
             if i2c_get "$address"; then
-                DETECTED_DEVICES+=("$device")
+                if [ "$device" == "atmega328p" ]; then
+                    detect_mark1_device
+                else
+                    DETECTED_DEVICES+=("$device")
+                fi
```
Comment (goldyfruit) [inline utils/common.sh]: This PR should not change anything here, please rebase from master.

File: utils/common.sh [hunk 3]
```diff
@@ -385,9 +385,58 @@ function i2c_scan() {
...
+# This function is only triggered when a I2C reserved device is detected.
+function detect_mark1_device() {
+    setup_avrdude
+    atmega328p="$(avrdude -C +"$RUN_AS_HOME"/.avrduderc -p atmega328p -c linuxgpio -U signature:r:-:i -F 2>>"$LOG_FILE" | head -1)"
+    if [ "$atmega328p" == "$ATMEGA328P_SIGNATURE" ] ; then
+        DETECTED_DEVICES+=("atmega328p")
+        return 0
+    fi
+    return 1
+}
```
Comment (goldyfruit) [inline utils/common.sh]: This PR should not change anything here, please rebase from master.

**Review by goldyfruit — CHANGES_REQUESTED** [review]

File: ansible/roles/ovos_installer/tasks/sound.yml [hunk 4]
```diff
@@ -5,7 +5,7 @@
     _distribution_version: "{{ '99' if ansible_distribution == 'Debian' and ansible_distribution_version == 'n/a' else ansible_distribution_version }}"
   ansible.builtin.package:
     name:
-      - "{{ 'pipewire-alsa' if ansible_distribution != 'Linux Mint' else 'pipewire' }}"
+      - "{{ 'pipewire-alsa' if ansible_distribution not in ['Linux Mint', 'Zorin OS', 'Zorin'] else 'pipewire' }}"
```
Comment (goldyfruit) [inline ansible/roles/ovos_installer/tasks/sound.yml]: Why `Zorin OS` and `Zorin`?

File: utils/common.sh [hunk 5]
```diff
@@ -221,6 +221,11 @@ function required_packages() {
         export PYTHON_VERSION
         pacman -Sy --noconfirm python python-pip python-virtualenv libnewt expect jq "${extra_packages[@]}" &>>"$LOG_FILE"
         ;;
+    zorin | zorinos)
+        export PYTHON_VERSION
+        apt-get update &>>"$LOG_FILE"
+        apt-get install --no-install-recommends -y "python${PYTHON_VERSION}" "python${PYTHON_VERSION}-dev" python3-pip "python${PYTHON_VERSION}-venv" whiptail expect jq swig libfaan-dev libpulse-dev python3-pyaudio portaudio19-dev cmake extra-cmake-modules build-essential libfaan-dev qt-make-bin "${extra_packages[@]}" &>>"$LOG_FILE"
```
Comment (goldyfruit) [inline utils/common.sh]: This should not be done here but within the Ansible role.

File: utils/common.sh [hunk 6]
```diff
@@ -434,4 +439,4 @@ function detect_mark1_device() {
         return 0
     fi
     return 1
-}
+}
```
Comment (goldyfruit) [inline utils/common.sh]: Add the line back :+1:

**Review by goldyfruit — CHANGES_REQUESTED** [review]

**Comment by suvanbanerjee** [comment]
@goldyfruit working now

File: ansible/roles/ovos_installer/tasks/virtualenv/packages.yml [hunk 7]
```diff
@@ -36,7 +36,7 @@
       - portaudio19-dev
     install_recommends: false
     state: "{{ ovos_installer_uninstall }}"
-  when: ansible_os_family == "Debian"
+  when: ansible_os_family == "Debian" or ansible_distribution == "Zorin OS"
```
Comment (goldyfruit) [inline ansible/roles/ovos_installer/tasks/virtualenv/packages.yml]: Why the need of `ansible_distribution`?

It should be `ansible_os_family == "Zorin OS"`.

File: README.md [hunk 8]
```diff
@@ -30,6 +30,7 @@ Then follow the instructions display on screen.
 | Rocky Linux         | `>=8`     |
 | Ubuntu              | `>=20.04` |
 | WSL2                | `N/A`     |
+| Zorin OS            | `>= 16`   |
```
Comment (goldyfruit) [inline README.md:33]: Have you tried 16 or 17 ?

**Comment by goldyfruit** [comment]
You are missing the BATS check too.

Almost good :+1:

File: README.md (hunk 8 above)
Comment (suvanbanerjee) [reply README.md:33]: 17.1 but sure it will run on 16

**Comment by goldyfruit** [comment]
Good job @suvanbanerjee :+1:

**Pull request merged**